From 7c35f7be17e3299f2c25c318f2e6a766b059ea52 Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Fri, 28 Dec 2018 08:37:44 +0000
Subject: [PATCH 1/2] ARM build tweaks

Signed-off-by: Edmunt Pienkowsky <roed@onet.eu>

diff --git a/builds/posix/prefix.linux_arm b/builds/posix/prefix.linux_arm
index b59d8a834c..73344e042a 100644
--- a/builds/posix/prefix.linux_arm
+++ b/builds/posix/prefix.linux_arm
@@ -18,10 +18,10 @@
 #	14 Apr 2008	Alan Barclay	alan AT escribe.co.uk
 
 
-#LD=@CXX@
+COMMON_FLAGS=-DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -DARM -MMD -fPIC -fmessage-length=0 -fno-delete-null-pointer-checks
+CXXFLAGS:=$(CXXFLAGS) -std=gnu++03
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer
+WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -Wno-narrowing
 
-#PROD_FLAGS=-ggdb -O3 -fno-omit-frame-pointer -DLINUX -pipe -MMD -fPIC
-#DEV_FLAGS=-ggdb -DLINUX -DDEBUG_GDS_ALLOC -pipe -MMD -p -fPIC -Wall -Wno-switch
-
-PROD_FLAGS=-O3 -DLINUX -DARM -pipe -p -MMD -fPIC -fsigned-char -fmessage-length=0 -DFB_SEND_FLAGS=MSG_NOSIGNAL
-DEV_FLAGS=-ggdb -DLINUX -DARM -pipe -p -MMD -fPIC -Wall -fsigned-char -fmessage-length=0 -Wno-non-virtual-dtor -DFB_SEND_FLAGS=MSG_NOSIGNAL
+PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
+DEV_FLAGS=$(COMMON_FLAGS) $(WARN_FLAGS) $(DEBUG_CXXFLAGS)
diff --git a/builds/posix/prefix.linux_arm64 b/builds/posix/prefix.linux_arm64
index 43043a071a..2384a43bb6 100644
--- a/builds/posix/prefix.linux_arm64
+++ b/builds/posix/prefix.linux_arm64
@@ -17,11 +17,10 @@
 # Start of file prefix.linux:  $(VERSION)  $(PLATFORM)
 #      14 Apr 2008     Alan Barclay    alan AT escribe.co.uk
 
+COMMON_FLAGS=-DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -DARM64 -MMD -fPIC -fsigned-char -fmessage-length=0 -fno-delete-null-pointer-checks
+CXXFLAGS:=$(CXXFLAGS) -std=gnu++03
+OPTIMIZE_FLAGS=-fno-omit-frame-pointer
+WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -Wno-narrowing
 
-#LD=@CXX@
-
-#PROD_FLAGS=-ggdb -O3 -fno-omit-frame-pointer -DLINUX -pipe -MMD -fPIC
-#DEV_FLAGS=-ggdb -DLINUX -DDEBUG_GDS_ALLOC -pipe -MMD -p -fPIC -Wall -Wno-switch
-
-PROD_FLAGS=-O3 -DLINUX -DARM64 -pipe -p -MMD -fPIC -fsigned-char -fmessage-length=0 -DFB_SEND_FLAGS=MSG_NOSIGNAL
-DEV_FLAGS=-ggdb -DLINUX -DARM64 -pipe -p -MMD -fPIC -Wall -fsigned-char -fmessage-length=0 -Wno-non-virtual-dtor -DFB_SEND_FLAGS=MSG_NOSIGNAL
+PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
+DEV_FLAGS=$(COMMON_FLAGS) $(WARN_FLAGS) $(DEBUG_CXXFLAGS)
-- 
2.20.1

