From 1d732735d0e3acb25a9800cddc7afbf1eb07528a Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Fri, 16 Aug 2019 12:48:39 +0200
Subject: [PATCH 4/7] Configuration tweaks

Use more secure options by default.

Signed-off-by: Edmunt Pienkowsky <roed@onet.eu>

diff --git a/builds/install/misc/databases.conf.in b/builds/install/misc/databases.conf.in
index 8292777bbd..4bbafa1b49 100644
--- a/builds/install/misc/databases.conf.in
+++ b/builds/install/misc/databases.conf.in
@@ -15,10 +15,14 @@
 #
 
 #
-# Example Database:
+# Your first database
 #
-employee.fdb = $(dir_sampleDb)/employee.fdb
-employee = $(dir_sampleDb)/employee.fdb
+my-first = /var/lib/firebird/data/my-first.fdb
+{
+	RemoteAccess = false
+	DefaultDbCachePages = 50
+	SecurityDatabase = my-first
+}
 
 #
 # Master security database specific setup.
diff --git a/builds/install/misc/firebird.conf.in b/builds/install/misc/firebird.conf.in
index 803103d549..c048a6d9a4 100644
--- a/builds/install/misc/firebird.conf.in
+++ b/builds/install/misc/firebird.conf.in
@@ -123,7 +123,7 @@
 #
 # Type: string (special format)
 #
-#DatabaseAccess = Full
+DatabaseAccess = None
 
 
 # ----------------------------
@@ -189,7 +189,7 @@
 #
 # Type: string (special format)
 #
-#UdfAccess = Restrict UDF
+UdfAccess = None
 
 
 # ----------------------------
-- 
2.22.0

