From 0ebbf1b5bfc88ed54ae7faea43f83cdcc92ac014 Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Tue, 1 Jan 2019 11:30:41 +0000
Subject: [PATCH 4/6] Configuration tweaks

Use more secure options by default.

Signed-off-by: Edmunt Pienkowsky <roed@onet.eu>

diff --git a/builds/install/misc/databases.conf.in b/builds/install/misc/databases.conf.in
index 8292777bbd..0ba8859120 100644
--- a/builds/install/misc/databases.conf.in
+++ b/builds/install/misc/databases.conf.in
@@ -15,19 +15,11 @@
 #
 
 #
-# Example Database:
+# Your first database:
 #
-employee.fdb = $(dir_sampleDb)/employee.fdb
-employee = $(dir_sampleDb)/employee.fdb
-
-#
-# Master security database specific setup.
-# Do not remove it until you understand well what are you doing!
-#
-security.db = $(dir_secDb)/security3.fdb
+my-first = /var/lib/firebird/my-first.fdb
 {
-	RemoteAccess = false
-	DefaultDbCachePages = 50
+	SecurityDatabase = my-first
 }
 
 #
diff --git a/builds/install/misc/firebird.conf.in b/builds/install/misc/firebird.conf.in
index 803103d549..0411a35588 100644
--- a/builds/install/misc/firebird.conf.in
+++ b/builds/install/misc/firebird.conf.in
@@ -123,7 +123,7 @@
 #
 # Type: string (special format)
 #
-#DatabaseAccess = Full
+DatabaseAccess = None
 
 
 # ----------------------------
@@ -176,10 +176,6 @@
 # are stored. Relative paths are treated relative to the root directory 
 # of firebird.
 #
-# Default value 'Restrict UDF' provides the same restrictions
-# as in FB 1.0. To specify access to specific trees, enum all required
-# paths (for Windows this may be something like 'C:\ExternalFunctions',
-# for unix - '/db/extern;/mnt/extern').
 @UDF_COMMENT@
 # NOTE: THE EXTERNAL FUNCTION ENGINE FEATURE COULD BE USED TO COMPROMISE
 # THE SERVER/HOST AS WELL AS DATABASE SECURITY!!
@@ -189,7 +185,7 @@
 #
 # Type: string (special format)
 #
-#UdfAccess = Restrict UDF
+UdfAccess = None
 
 
 # ----------------------------
-- 
2.20.1

